ssz/src/main/java/org/ethereum/beacon/ssz/SSZSerializer.java

```diff
@@ -61,13 +59,10 @@ public SerializerResult visitList(SSZListType descriptor, Object listValue, int
    * @return deserialized instance of clazz or throws exception
    */
   public <C> C decode(byte[] data, Class<? extends C> clazz) {
-    DecodeResult decodeResult = sszVisitorHost.handleAny(
+    Object decodeResult = sszVisitorHost.handleAny(
         typeResolver.resolveSSZType(new SSZField(clazz)),
-        Pair.with(Bytes.wrap(data), null),
+        Bytes.wrap(data),
         new SosDeserializer());
-    if (data.length != decodeResult.readBytes) {
-      throw new SSZSerializeException(String.format("Invalid SSZ encoding, calculated data size %s bytes, while provided %s bytes", decodeResult.readBytes, data.length));
```

Comment:
Do I understand correctly that with addition of union to ssz, we couldn't keep this check at all?

Reply:
This is about sos not about unions. When decoding the end of the latest variable part is determined from the byte array length. So if we have erroneous extra bytes, then they either would be decoded to erroneous extra list elements (and this can't be detected) or the fail would happen during decoding of fixed size value

Reply:
Got it